src/components/SnackChat/SnackChat.js

```diff
@@ -242,114 +252,159 @@ class SnackChat extends Component {
       const oldestPosition = this.positionBuffer[this.i];
       forEachAttribute(
         (bodyPart, attribute) =>
-          (this.averageBodyPosition[bodyPart][attribute] =
-            (this.averageBodyPosition[bodyPart][attribute] *
-              POSITION_BUFFER_SIZE -
+          (averageBodyPosition[bodyPart][attribute] =
+            (averageBodyPosition[bodyPart][attribute] * POSITION_BUFFER_SIZE -
               oldestPosition[bodyPart][attribute] +
               body[bodyPart][attribute]) /
             POSITION_BUFFER_SIZE)
       );
     }
     this.positionBuffer[this.i] = body;
-
-    // Video background
-    this.ctx.save();
-    this.drawBackground(video);
-
-    // Filter
-    const shoulders = this.averageBodyPosition.shoulders;
-    this.ctx.save();
-    this.ctx.rotate(shoulders.angle);
-    this.ctx.drawImage(
-      this.filter,
-      shoulders.rightX -
-        shoulders.span * 1.5 +
-        shoulders.span * shoulders.angle,
-      shoulders.rightY - shoulders.span * 1.5,
-      shoulders.span * 4,
-      shoulders.span * 4
-    );
-    this.ctx.restore();
-
-    // Clip face
-    const ears = this.averageBodyPosition.ears;
-    this.ctx.save();
-    this.ctx.translate(
-      ears.rightX + ears.width / 2,
-      ears.rightY + ears.height * ears.angle
-    );
-    this.ctx.rotate(ears.angle);
-    clipEllipse(this.ctx, 0, 0, ears.span * 1.5, ears.span * 1.5);
-    this.ctx.resetTransform();
-
-    // Re-draw face
-    this.drawBackground(video);
+    this.setState({
+      averageBodyPosition
+    });
 
     requestAnimationFrame(this.update);
   };
 
   onConnect = () => {
     this.ctx = this.canvas.current.getContext('2d');
-    this.ctx.lineWidth = 5;
-    this.ctx.strokeStyle = 'red';
-    this.filter = new Image();
-    this.filter.src = this.props.storeList[this.props.actualItem].image;
+    this.filter = this.props.storeList[this.props.actualItem].image;
     this.countdown();
     this.playGettingInPositionAnimation();
-    requestAnimationFrame(this.update);
   };
 
   onFail = () => {
     this.props.setSendWithPhoto(false);
     this.props.history.replace('/slackname');
   };
 
+  // render functions ================================================
```

Comment:
Since all the methods are called render we can find them quite easily searching for "render" so we can probably remove this :-)